components/brave_ads/rust/parse_and_sanitize_html_util.cc

```diff
@@ -0,0 +1,16 @@
+/* Copyright (c) 2023 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include "brave/components/brave_ads/rust/parse_and_sanitize_html_util.h"
+
+#include "brave/components/brave_ads/rust/lib.rs.h"
+
+namespace brave_ads::rust {
+
+std::string ParseAndSanitizeOgTagAttribute(const std::string& html) {
+  return static_cast<std::string>(parse_and_sanitize_og_tag_attribute(html));
```

Comment:
why do we have this static cast? Doesn't returning `CxxString` give us std::string libs.rs.h? These wrappers wouldn't be needed at all in that case.

Reply:
I asked about this earlier. The rust functions return a rust `std::String` which cxx turns into a `rust::String` on the C++ side. The static cast is supposed to trigger the `std::string` implicit conversion operator to create a new `std::string`. I assume this copies, but keeps the allocations isolated on each side of the ffi while letting both languages use their standard types.

Reply:
yep, it calls a `std::string` cast operator:
https://github.com/dtolnay/cxx/blob/1.0.82/src/cxx.cc#L316

Reply:
yes, but that's because we're returning rust::String instead of CxxString. See skus
```    
pub struct HttpResponse<'a> {
        pub result: SkusResult,
        pub return_code: u16,
        pub headers: &'a CxxVector<CxxString>,
        pub body: &'a CxxVector<u8>,
```

which converts directly to `std::vector<std::string>` with no  cast

Reply:
hmm, if I remove static_cast, then compilation fails with:

```
error: no viable conversion from returned value of type '::rust::String' to function return type 'std::string' (aka 'basic_string<char>')
  return strip_non_ascii_characters(content);
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../../buildtools/third_party/libc++/trunk/include/string:852:33: note: candidate constructor not viable: no known conversion from '::rust::String' to 'const string &' for 1st argument
  _LIBCPP_CONSTEXPR_SINCE_CXX20 basic_string(const basic_string& __str)
                                ^
```

Reply:
The suggestion was to also change the return value of `string_non_ascii_characters` to `cxx::CxxString` and do the conversion in Rust at the end of that function.

Reply:
ok, I will use cxx::CxxString